File: p9_suma_matrices.f90
PROGRAM SUMA_MATRICES
  IMPLICIT NONE
  INTEGER :: R, C, I, J
  INTEGER, DIMENSION(20,20) :: A, B, S

  PRINT *, 'Ingrese filas y columnas (r c, max 20 20):'
  READ *, R, C

  IF (R <= 0 .OR. C <= 0 .OR. R > 20 .OR. C > 20) THEN
     PRINT *, 'Error: dimensiones deben ser positivas y <= 20'
     STOP
  END IF

  PRINT *, 'Ingrese la matriz A:'
  DO I = 1, R
     PRINT *, 'Fila ', I, ':'
     DO J = 1, C
        READ *, A(I,J)
     END DO
  END DO

  PRINT *, 'Ingrese la matriz B:'
  DO I = 1, R
     PRINT *, 'Fila ', I, ':'
     DO J = 1, C
        READ *, B(I,J)
     END DO
  END DO

  DO I = 1, R
     DO J = 1, C
        S(I,J) = A(I,J) + B(I,J)
     END DO
  END DO

  PRINT *, 'Resultado S = A + B:'
  DO I = 1, R
     DO J = 1, C
        WRITE(*, '(I8)', ADVANCE='NO') S(I,J)
     END DO
     PRINT *
  END DO

  PRINT *, 'Presione ENTER para finalizar'
  READ *

END PROGRAM SUMA_MATRICES

File: p9_suma_matrices.pas
program SumaMatrices;
uses crt;

const
  MAX = 20;  
var
  r, c, i, j: integer;
  A, B, d: array[1..MAX, 1..MAX] of integer;

begin
  clrscr;
  write('Ingrese filas y columnas (r c, max ', MAX, '): ');
  readln(r, c);

  if (r <= 0) or (c <= 0) or (r > MAX) or (c > MAX) then
  begin
    writeln('Dimensiones no validas. Deben ser entre 1 y ', MAX);
    halt(0);
  end;

  writeln('Ingrese la matriz A:');
  for i := 1 to r do
    for j := 1 to c do
    begin
      write('A[', i, ',', j, ']= ');
      readln(A[i, j]);
    end;

  writeln('Ingrese la matriz B:');
  for i := 1 to r do
    for j := 1 to c do
    begin
      write('B[', i, ',', j, ']= ');
      readln(B[i, j]);
    end;

  for i := 1 to r do
    for j := 1 to c do
      d[i, j] := A[i, j] + B[i, j];

  writeln;
  writeln('Resultado C = A + B:');
  for i := 1 to r do
  begin
    for j := 1 to c do
      write(d[i, j]:8);
    writeln;
  end;

  readln;
end.
